src/main/scala/chisel3/util/circt/IsX.scala

```diff
@@ -0,0 +1,37 @@
+// SPDX-License-Identifier: Apache-2.0
+
+package chisel3.util.circt
+
+import chisel3._
+import chisel3.experimental.{annotate, ChiselAnnotation, ExtModule}
+import chisel3.internal.Builder
+
+import circt.Intrinsic
+
+/** Create a module with a parameterized type which returns whether the input
+  * is a verilog 'x'.
+  */
+private class IsXIntrinsic[T <: Data](gen: T) extends ExtModule {
+  val i = IO(Input(gen))
+  val found = IO(Output(UInt(1.W)))
+  annotate(new ChiselAnnotation {
+    override def toFirrtl =
+      Intrinsic(toTarget, "circt.isX")
+  })
+  override val desiredName = "IsX_" + Builder.idGen.next
```

Comment:
Don't use `idGen`, as you note in the tests below, it is shared between Chisel runs. You should do basically exactly what https://github.com/chipsalliance/chisel3/pull/2717 does in it's changes to the Builder. Add a new `mutable.HashMap[String, Long]` which would let you have a separate namespace for each of these ExtModules.

Then in the function you put in `Builder`, you can have that return the fresh id for the particular namespace and update the underlying mutable HashMap.